dbms/src/Functions/FunctionsIsIPAddr.h

```diff
@@ -0,0 +1,259 @@
+
+    /* Description：
+    * This function is used to determine whether the input string is an IPv6 address, 
+    * and the code comes from the inet_pton6 function of "arpa/inet.h".
+    * References: http://svn.apache.org/repos/asf/apr/apr/trunk/network_io/unix/inet_pton.c
+    */
+    static inline UInt8 isMatch(const char * src)
+    {
+        if (nullptr == src)
+            return 0;
+        static const char xdigits_l[] = "0123456789abcdef",
+                          xdigits_u[] = "0123456789ABCDEF";
+        unsigned char tmp[16], *tp, *endp, *colonp;
+        const char *xdigits, *curtok;
+        int ch, saw_xdigit;
+        unsigned int val;
+
+        memset((tp = tmp), '\0', IN6ADDRSZ);
+        endp = tp + IN6ADDRSZ;
+        colonp = nullptr;
+        if (*src == ':')
+            if (*++src != ':')
+                return 0;
+        curtok = src;
+        saw_xdigit = 0;
+        val = 0;
+        while ((ch = *src++) != '\0')
+        {
+            const char * pch;
+
+            if ((pch = strchr((xdigits = xdigits_l), ch)) == nullptr)
+                pch = strchr((xdigits = xdigits_u), ch);
+            if (pch != nullptr)
+            {
+                val <<= 4;
+                val |= (pch - xdigits);
+                if (val > 0xffff)
+                    return 0;
+                saw_xdigit = 1;
+                continue;
+            }
+            if (ch == ':')
+            {
+                curtok = src;
+                if (!saw_xdigit)
+                {
+                    if (colonp)
+                        return 0;
+                    colonp = tp;
+                    continue;
+                }
+                if (tp + INT16SZ > endp)
+                    return 0;
+                *tp++ = static_cast<unsigned char>(val >> 8) & 0xff;
+                *tp++ = static_cast<unsigned char>(val) & 0xff;
+                saw_xdigit = 0;
+                val = 0;
+                continue;
+            }
+            if (ch == '.' && ((tp + INADDRSZ) <= endp) && CheckIsIPv4Impl::isMatch(curtok) > 0)
+            {
+                tp += INADDRSZ;
+                saw_xdigit = 0;
+                break; /* '\0' was seen by CheckIsIPv4Impl::isMatch(). */
+            }
+            return 0;
+        }
+        if (saw_xdigit)
+        {
+            if (tp + INT16SZ > endp)
+                return 0;
+            *tp++ = static_cast<unsigned char>(val >> 8) & 0xff;
+            *tp++ = static_cast<unsigned char>(val) & 0xff;
+        }
+        if (colonp != nullptr)
+        {
+            const size_t n = tp - colonp;
+            size_t i;
+
+            for (i = 1; i <= n; ++i)
+            {
+                endp[-i] = colonp[n - i];
+                colonp[n - i] = 0;
+            }
+            tp = endp;
+        }
+        if (tp != endp)
+            return 0;
+        return 1;
+    }
+};
+
+template <typename Impl>
+class FunctionIsIPv4OrIsIPv6 : public IFunction
```

Comment:
From https://github.com/pingcap/tidb/blob/6d8669e9875d66f84920d8505bc0b6927479df02/expression/builtin_miscellaneous_vec.go#L139-L142, we shoud use `bool useDefaultImplementationForNulls() const override { return false; }` here and handle nullable data type.

Reply:
done，Sorry, I forgot to handle the nullable case.